geoapps_utils/driver/data.py

```diff
@@ -65,10 +65,13 @@ def collect_input_from_dict(
                 and not isinstance(info.annotation, GenericAlias)
                 and issubclass(info.annotation, BaseModel)
             ):
-                update[field] = BaseData.collect_input_from_dict(
-                    info.annotation,
-                    data,  # type: ignore
-                )
+                if field in data and isinstance(data[field], info.annotation):
+                    update[field] = data[field]
```

Comment:
Same operation as what is happening in the logic below. Can we combine?

Reply:
You can't combine the `if field in data` conditions because they're on different sides of the outer if/else.  One is for when info.annotation is a BaseData subclass, and the other is for plain annotations.